elijah/README.md

```diff
@@ -0,0 +1,305 @@
+
+As an example, the result of `letterTriangle(4)` will look like this when printed:
+```
+A
+BA
+CBA
+DCBA
+```
+
+    var letterTriangle = function(n) {
+        // alphabet
+        var alph = ['A','B','C','D','E','F','G','H','I','J',
+                'K','L','M','N','O','P','Q','R','S','T',
+                'U','V','W','X','Y','Z']; 
+        var tri = '';
+        var count;
+        
+        for (i = 0; i < n; i++) {
+            // set count to i, share with inner loop
+            count = i;
+            for (count; count >= 0; count--) {
+                tri += alph[count];
+            }
+            tri += '\n';
+        }
+        return tri;
+    }
+
+
+
+**b)**
+Adapting your solution to part a), write a function which returns a string containing the entire lyrics for the song "The Twelve Days of Christmas".  Make sure that your result is grammatically and typographically correct (include line breaks, commas, etc. where needed), but keep redundancy within your program to a minimum.  Don't just `console.log()` each line; as before, return them together as a single string which includes a '\n' at each line break.
+You may want to use helper functions, loops, and/or arrays to store repeated elements.
+
+If you prefer a non-Christmas option, you may choose a different song with similarly repeating structure, such as "There was an Old Woman Who Swallowed a Fly".  If you prefer a vegan option, you may write your own cruelty-free song.
+
+    var spongeBob = "Spongebob Squarepants!";
+    var outputSong = '';
+    var song = ['Are you ready kids?\n "Aye Aye Captain"\nI can\'t hear you!\n "AYE AYE CAPTAIN"\nOohh...\nWho lives in a pineapple under the sea?','Absorbant and yellow and porous is he!','If nautical nonsense be something you wish','Then drop on the deck and flop like a fish!','READY?'];
+    
+    var sponge = function(s) {
+        // take in string and append "spongebob squarepants!"
+        return s = '\n' + spongeBob + '\n';
+    }
+    
+    var songBuilder = function(a) {
+        // build song string from array a
+        for (i = 0; i < a.length; i++) {
+            outputSong += a[i];
+            outputSong += sponge(outputSong);
+        }
+        for (i = 0; i < 2; i++) {
+            outputSong += sponge(outputSong);
+        }
+        return outputSong += '\n' + spongeBob.toUpperCase();
+    }
+
+    // "play" song
+    console.log(songBuilder(song));
+
+---
+
+**4)** _(Difficult, 20%)_
+
+Suppose the '&' key on your keyboard is missing, and you want to be able to continue programming without it.  You decide to simulate the && operator with a function!
+
+**a)**
+Write a function `and2(a,b)` which tries to simulate the && operator: it should always return the same result as `(a && b)` for any values of _a_ and _b_.  (For example, `and2((0>1),true)` should return _false_.)  But you can't use && itself within your function!
+
+
+    var and2 = function(a,b) {
+        return a ? (b ? true : false) : false
+    }
+
```

Comment:
Yep! Simplified, you could write it like this: 

function and2(a,b) {
	return (a? b: a);  //ternary conditional operator
}